docs/soci.md

```diff
@@ -0,0 +1,29 @@
+# Lazy-pulling using SOCI Snapshotter
+
+| :zap: Requirement | nerdctl >= 1.1.1 |
+| ----------------- |------------------|
+
+SOCI Snapshotter is a containerd snapshotter plugin. It enables standard OCI images to be lazily loaded without requiring a build-time conversion step. "SOCI" is short for "Seekable OCI", and is pronounced "so-CHEE".
+
+See https://github.com/awslabs/soci-snapshotter to learn further information.
+
+## Enable SOCI for `nerdctl run`
+
+- Install containerd remote snapshotter plugin (`soci`) from https://github.com/awslabs/soci-snapshotter/blob/main/docs/GETTING_STARTED.md#prerequisites
+
+- Add the following to `/etc/containerd/config.toml`:
+```toml
+[proxy_plugins]
+  [proxy_plugins.soci]
+    type = "snapshot"
+    address = "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+```
+
+- Launch `containerd` and `soci-snapshotter`
+
+- Run `nerdctl` with `--snapshotter=soci`
+```console
+nerdctl run --it --rm --snapshotter=soci docker.io/library/alpine:3.14.2
+```
+
+For more details about how to build soci image, please refer to [soci-snapshotter](https://github.com/awslabs/soci-snapshotter/blob/main/docs/GETTING_STARTED.md#create-and-push-a-soci-index)
```

Comment:
For building SOCI index, do you think we can add a --snapshotter=soci in `nerdctl build` to cover it? As we are **integrating** SOCI to nerdctl, I feel it can provide better user experience if we can let them completely use nerdctl CLI instead of SOCI CLI.

Reply:
> For building SOCI index, do you think we can add a --snapshotter=soci in nerdctl build to cover it? 

Yes, but difficult

Reply:
Would it just be a simple `soci create` command call ([doc](https://github.com/awslabs/soci-snapshotter/blob/main/docs/getting-started.md#create-soci-index)) in soci snapshotter branch at the end of build command logic? https://github.com/containerd/nerdctl/blob/ba23d5d7e5eda4d3de739edcbf99ce633c0433f2/pkg/cmd/builder/build.go#L117